script/tool/test/analyze_command_test.dart

```diff
@@ -53,6 +53,24 @@ void main() {
         ]));
   });
 
+  test('skips flutter pub get for examples', () async {
+    final Directory plugin1Dir = createFakePlugin('a', withSingleExample: true);
+
+    final MockProcess mockProcess = MockProcess();
+    mockProcess.exitCodeCompleter.complete(0);
+    processRunner.processToReturn = mockProcess;
+    await runner.run(<String>['analyze']);
+
+    expect(
+        processRunner.recordedCalls,
+        orderedEquals(<ProcessCall>[
+          ProcessCall(
+              'flutter', const <String>['packages', 'get'], plugin1Dir.path),
+          ProcessCall('dart', const <String>['analyze', '--fatal-infos'],
+              plugin1Dir.path),
+        ]));
+  });
+
```

Comment:
Could you also add a test of a package with a sub-package that's not `example`, to make sure nobody makes the "simplification" that I was about to suggest before checking flutter/packages?

Reply:
We have one like that already - the `analyzes all packages` test - but I can add one where the 2nd package is named 'example' but not contained by the 1st package.